File: javascript/hello.js
console.log("hello world");

// // Variables
// // let, var, const
let colour = "green";
console.log(`My favourite ${colour}`);

// // Numbers
let boxes = 7;
console.log(`Boxes: ${boxes}.`);

let weight = 1.5;
console.log(`Weight: ${weight}.`);

let total = weight * boxes;
console.log(`Total: ${total}.`);

// // Booleans
let isTrue = true;
let isFalse = false;

// Array
let animals = ['monkey', 'fox', 'sugar glider'];
console.log(`${animals}`);

// For loops
for (let animal of animals) {
    console.log(`Animal is ${animal}`)
}

// Else if
for (let animal of animals) {
    console.log(`The ${animal} goes...`);
    if (animal == 'monkey') {
        console.log('oooaaa');
    } else if (animal === 'fox') {
        console.log('ding ding ding');
    } else if (animal === 'sugar glider') {
        console.log('woof!');
    }
}

// Objects vs hashes
let progress = {
    program: "Web development",
    language: "Javascript",
    is_interested: true
};

console.log(progress)

// Function
function greeting(name) {
    console.log(`Welcome back ${name}!`);
}

greeting("Hunter");
greeting("Hardeep");
greeting("Anthony");

File: ruby/hello.rb
puts "hello world"

# # Variables
colour = "green"
puts "My favourite color #{colour}."

# #  Numbers
boxes = 7
puts "Boxes: #{boxes}"

weight = 1.5;
puts "Weight: #{weight}"

total = weight * boxes
puts "Total: #{total}"

# # Booleans
isTrue = true
isFalse = false
# comment out:cmd+/

# Array 
animals = ['monkey', 'fox', 'sugar glider']
# puts "#{animals}"

# For loops
for animal in animals do
    puts "Animal is #{animal}"
end

# Else if
for animal in animals do
    puts "The #{animal} goes..."
    if (animal == 'monkey')
        puts 'oooaaa'
    elsif (animal == 'fox')
        puts 'ding ding ding'
    elsif (animal == 'sugar glider')
        puts 'hiss'
    end
end

# Hashes
progress = {
    program: "Web development",
    language: "Ruby",
    is_interested: false
}

puts progress

# Function
# A function is a reusable set of instructions
def greeting(name)
    puts "Goodbye, #{name}!"
    return
end

greeting("Bianca")
greeting("Hunter")
greeting("Hardeep")
greeting("Anthony")